operator/identitygc/crd_gc.go

```diff
@@ -170,9 +170,11 @@ func (igc *GC) gc(ctx context.Context) error {
 	if ctx.Err() == nil {
 		igc.successfulRuns++
 		igc.metrics.IdentityGCRuns.WithLabelValues(LabelValueOutcomeSuccess).Set(float64(igc.successfulRuns))
+		igc.metrics.IdentityGCLatency.Set(float64(time.Since(timeNow).Seconds()))
 	} else {
 		igc.failedRuns++
 		igc.metrics.IdentityGCRuns.WithLabelValues(LabelValueOutcomeFail).Set(float64(igc.failedRuns))
+		igc.metrics.IdentityGCLatency.Set(float64(0))
```

Comment:
In case of failure we're resetting the gauge to 0. I think it would be better to use `metric.Vec[metric.Gauge]` (just like `IdentityGCRuns`) in order to attach label values like `LabelValueOutcomeSuccess` and `LabelValueOutcomeFail`).
This would allow for more fine-grained analysis, e.g: "how is the identity GC latency when the GC does not fail?"